hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/async/RouterAsyncClientProtocol.java

```diff
@@ -1057,33 +1057,33 @@ public boolean setReplication(String src, short replication)
    *         subclusters else false in all other scenarios.
    * @throws IOException if unable to get the file status.
    */
+  @VisibleForTesting
   @Override
   public boolean isMultiDestDirectory(String src) throws IOException {
-    try {
+    asyncComplete(false);
```

Comment:
Hi, @hfutatzhanghb don’t need to add this here. If the code inside the **asyncTry** block executes without any exceptions, a **CompletableFuture** containing **true** or **false** will definitely be set to the thread-local variable. If the **asyncTry** block throws an **UnresolvedPathException**, it can be caught by **asyncCatch** and a **CompletableFuture** containing **false** will be set to the thread-local variable. If the **asyncTry** block throws any **other exception**, a **CompletableFuture** containing that exception will be set to the thread-local variable for the outer method to handle, which is consistent with the original synchronous logic.

Reply:
@KeeProMise very nice suggestion, thanks for reviewing. have fixed.